coderd/users.go

```diff
@@ -171,16 +171,38 @@ func (api *API) postFirstUser(rw http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	//nolint:gocritic // needed to create first user
+	defaultOrg, err := api.Database.GetDefaultOrganization(dbauthz.AsSystemRestricted(ctx))
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error fetching default organization. If you are encountering this error, you will have to restart the Coder deployment.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	//nolint:gocritic // ensure everyone group
+	_, err = api.Database.InsertAllUsersGroup(dbauthz.AsSystemRestricted(ctx), defaultOrg.ID)
```

Comment:
I thought about that too. What I didn't like about the migration was I can't tell if the `INSERT INTO organizations` ran with an insert, or without. So the followup of `INSERT INTO groups` cannot be dependent on if an org did not exist at the start of the migration or not.

Putting it in `postFirstUser` ensures that it is a new instance setup, so I know this is safe.

Although we do not allow dropping the everyone group, so it _should_ be a safe `INSERT INTO groups ... DO NOTHING`. 

---

I can move it to a migration if you feel that is better.

Reply:
I'm just wary of all this side-effect stuff of creating a user.

I also kind of like the symmetry of creating an org and the corresponding everyone group together, since that's presumably what we'll do when we allow creating new organizations, since there is no similar "first user creation" event to tie it to.

Reply:
I saw this comment after the approval merge, my mistake.

I can make another migration for the everyone group and do nothing on conflict.